tests/testImageMapReduce.py

```diff
@@ -167,71 +166,113 @@ def testExampleTaskWithOverlaps(self):
+                            config.reducerSubtask.reduceOperation = reduceOp
+                            n_tests += 1
+                            self._testGridValidity(config, gstepx, gsizex, gstepy, gsizey, expectedVal)
+        print("Ran a total of %d grid validity tests." % n_tests)
+
+    def _testGridValidity(self, config, gstepx, gsizex, gstepy, gsizey, expectedVal=1.):
+        """Method to test the grid validity given an input config.
+
+        Here we also iterate over scaleByFwhm in (True, False) and
+        ensure that we get more `boxes` when `scaleByFwhm=False` than
+        vice versa.
+
+        Parameters
+        ----------
+        config : `ipDiffim.AddAmountImageMapReduceConfig`
+            input AddAmountImageMapReduceConfig
+        gstepx : `float`
+            grid x-direction step size
+        gsizex : `float`
+            grid x-direction box size
+        gstepy : `float`
+            grid y-direction step size
+        gsizey : `float`
+            grid y-direction box size
+        expectedVal : `float`
+            float to add to exposure (to compare for testing)
+        """
         config.mapperSubtask.addAmount = expectedVal
-
-        for scaleByFwhm in (False, True):
+        lenBoxes = [0, 0]
+        for scaleByFwhm in (True, False):
             config.scaleByFwhm = scaleByFwhm
-            for gsx in range(51, 1, -10):
-                config.gridStepX = config.gridSizeX = gsx
-                for gsy in range(51, 1, -10):
-                    config.gridStepY = config.gridSizeY = gsy
-
-                    task = ImageMapReduceTask(config)
-                    boxes = task._generateGrid(self.exposure)
-                    if len(boxes[0]) > 1000:  # bypass to prevent slow testing
-                        continue
-                    newExp = task.run(self.exposure)
-                    newMI = newExp.getMaskedImage()
-                    newArr = newMI.getImage().getArray()
-                    mi = self.exposure.getMaskedImage()
-
-                    mn = newArr.mean()
-                    self.assertClose(mi.getImage().getArray().mean(), mn - expectedVal)
-                    self.assertClose(mn, expectedVal)
+            if scaleByFwhm:
+                config.gridStepX = gstepx
+                config.gridSizeX = gsizex
+                config.gridStepY = gstepy
+                config.gridSizeY = gsizey
+            else:  # otherwise the grid is too fine and elements too small.
+                config.gridStepX = gstepx * 3.
+                config.gridSizeX = gsizex * 3.
+                config.gridStepY = gstepy * 3.
+                config.gridSizeY = gsizey * 3.
+            task = ImageMapReduceTask(config)
+            boxes = task._generateGrid(self.exposure)
+            ind = 0 if scaleByFwhm else 1
+            lenBoxes[ind] = len(boxes[0])
+            if len(boxes[0]) > 1000:  # bypass to prevent slow testing
```

Comment:
I still don't like needing this if statement, particularly if you've reworked it to not result in such a large number of boxes.